src/common/type/list.c

```diff
@@ -294,9 +295,19 @@ lstInsert(List *this, unsigned int listIdx, const void *item)
                 this->listSizeMax *= 2;
                 this->list = memResize(this->list, this->listSizeMax * this->itemSize);
             }
+
+            this->listAlloc = this->list;
         }
         MEM_CONTEXT_END();
     }
+    // Else if there is space before the beginning of the list then move the list down
+    else if (
+        (this->list != this->listAlloc) &&
+        (this->listSize + ((uintptr_t)(this->list - this->listAlloc) / this->itemSize) == this->listSizeMax))
+    {
+        memmove(this->listAlloc, this->list, this->listSize * this->itemSize);
+        this->list = this->listAlloc;
+    }
 
     // If not inserting at the end then move items down to make space
```

Comment:
Could you move this comment  `// If not inserting at the end then move items down to make space` to right before the `if (listIdx != lstSize(this))` statement?

Reply:
Sure. Also added a comment.

Reply:
Awesome.